src/Lumper.UI/Updater/Updater.cs

```diff
@@ -0,0 +1,218 @@
+            if (match.Success)
+            {
+                MatchCollection currentVersion = Regex.Matches(match.ToString(), "[0-9]+");
+
+
+                if (int.TryParse(currentVersion[0].ToString(), out version.major)
+                    && int.TryParse(currentVersion[1].ToString(), out version.minor)
+                    && int.TryParse(currentVersion[2].ToString(), out version.patch))
+                {
+                }
+                else
+                {
+                    throw new Exception("Could not parse Major/Minor/Patch version.");
+                }
+
+            }
+            return version;
+
+        }
+        static void ExecuteCommand(string command)
+        {
+            int exitCode;
+            ProcessStartInfo processInfo = new ProcessStartInfo();
+            Process process;
+
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+            {
+                processInfo = new ProcessStartInfo("cmd.exe", "/c " + command);
+                processInfo.CreateNoWindow = true;
+                processInfo.UseShellExecute = false;
+            }
+            else if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
+            {
+                processInfo = new ProcessStartInfo(command);
+                processInfo.CreateNoWindow = true;
+                processInfo.UseShellExecute = true;
+            }
+            processInfo.RedirectStandardError = true;
+            processInfo.RedirectStandardOutput = true;
+            processInfo.WorkingDirectory = System.IO.Directory.GetCurrentDirectory();
+            process = Process.Start(processInfo);
+        }
+
+        //Grab the JSON from the github API and deserialize
+        private static async Task<GHUpdate> GetGithubUpdates()
+        {
+            HttpClient client = new HttpClient();
+
+            var request = new HttpRequestMessage()
+            {
+                RequestUri = new Uri("https://api.github.com/repos/momentum-mod/lumper/releases"),
+                Method = HttpMethod.Get
+            };
+            client.DefaultRequestHeaders.Add("User-Agent", "Other");
+            HttpResponseMessage m = client.Send(request);
+
+            if (m.IsSuccessStatusCode)
+            {
+                string response = await m.Content.ReadAsStringAsync();
+
+                var assets = JsonConvert.DeserializeObject<GHUpdate>(JsonObject.Parse(response)[0].ToString());
+                return assets;
+            }
+            else
+                throw new Exception("Could not connect - error " + m.StatusCode);
+        }
+        
+        public static async Task<bool> CheckForUpdate()
+        {
+            GHUpdate assets = await GetGithubUpdates();
+
+            //parse tag name to find the current and latest version
+            //finding the format of xx.yy.zz
+            string newestVersionSplit = Regex.Match(assets.tag_name, "[0-9]+\\.[0-9]+\\.[0-9]+").ToString();

```

Comment:
Okay, let's not do the `^` and `$` around the GeneratedRegex, then we can reuse it here though. Capture group stuff should mean they're not needed in `Version.Parse`